roles/ceph-docker-common/tasks/fetch_image.yml

```diff
@@ -1,16 +1,186 @@
 ---
+# NOTE (leseb): we must check each inventory group so this will work with collocated daemons
+- name: inspect ceph mon container
+  command: "docker inspect {{ ceph_mon_container_stat.stdout }}"
+  changed_when: false
+  register: ceph_mon_inspect
+  when:
+    - mon_group_name in group_names
+    - ceph_mon_container_stat.get('rc') == 0
+    - ceph_mon_container_stat.get('stdout_lines', [])|length != 0
+
+- name: inspect ceph osd container
+  command: "docker inspect {{ item }}"
+  changed_when: false
+  register: ceph_osd_inspect
+  with_items:
+    - "{{ ceph_osd_container_stat.stdout_lines | default([]) }}"
```

Comment:
Where is ``ceph_osd_container_stat`` created? I don't see that it's ever created in the CI logs. See: https://2.jenkins.ceph.com/job/ceph-ansible-prs-luminous-ansible2.4-docker_cluster/430/consoleFull

When an ansible task has a ``with_items`` value that is empty, like ``[]``, that task is not run at all so no vars that it registers gets set. You can see from the logs there is no status here returned at all from ansible on if the task failed, skipped, changed or was ok.

```
TASK [ceph-docker-common : inspect ceph osd container] *************************
task path: /home/jenkins-build/build/workspace/ceph-ansible-prs-luminous-ansible2.4-docker_cluster/roles/ceph-docker-common/tasks/fetch_image.yml:12
Monday 12 February 2018  18:17:52 +0000 (0:00:00.034)       0:05:37.770 ******* 
```

Because ``ceph_osd_inspect`` is never registered it causes every task below that needs it to not run and eventually fails with ``'ceph_osd_image_repodigest_before_pulling' is undefined``.

Reply:
https://github.com/ceph/ceph-ansible/blob/restart-if-new-img/roles/ceph-defaults/tasks/check_running_containers.yml#L13